src/Scaffolding/VS.Web.CG.Mvc/Minimal Api/MinimalApiGenerator.cs

```diff
@@ -0,0 +1,314 @@
+            }
+
+            List<string> errorList = new List<string>();
+
+            //check namespace for the Endpoints class for invalid keywords 
+            if (!string.IsNullOrEmpty(model.EndpointsNamespace) &&
+                !RoslynUtilities.IsValidNamespace(model.EndpointsNamespace))
+            {
+                errorList.Add(string.Format(
+                    CultureInfo.CurrentCulture,
+                    MessageStrings.InvalidNamespaceName,
+                    model.EndpointsNamespace));
+            }
+
+            if (!string.IsNullOrEmpty(model.EndpintsClassName) &&
+                !RoslynUtilities.IsValidIdentifier(model.EndpintsClassName))
+            {
+                errorList.Add(string.Format(
+                    CultureInfo.CurrentCulture,
+                    MessageStrings.InvalidClassName,
+                    model.EndpintsClassName));
+            }
+
+            if (errorList.Any())
+            {
+                throw new InvalidOperationException(string.Join("\n", errorList.ToArray()));
+            }
+        }
+
+        internal async Task ModifyProgramCs(string mapMethodName, string endpointsNamespace)
+        {
+            var jsonText = GetMinimalApiCodeModifierConfig();
+            CodeModifierConfig minimalApiChangesConfig = JsonSerializer.Deserialize<CodeModifierConfig>(jsonText);
+            if (minimalApiChangesConfig != null)
+            {
+                var programCsFile = minimalApiChangesConfig.Files.FirstOrDefault();
+                programCsFile.Usings = new string[] { endpointsNamespace };
+                var programType = ModelTypesLocator.GetType("<Program>$").FirstOrDefault() ?? ModelTypesLocator.GetType("Program").FirstOrDefault();
+                var project = Workspace.CurrentSolution.Projects.FirstOrDefault(p => p.AssemblyName.Equals(ProjectContext.AssemblyName, StringComparison.OrdinalIgnoreCase));
+                var programDocument = GetUpdatedDocument(project, programType);
+                var docEditor = await DocumentEditor.CreateAsync(programDocument);
+
+                var docRoot = docEditor.OriginalRoot as CompilationUnitSyntax;
+                var docBuilder = new DocumentBuilder(docEditor, programCsFile, ConsoleLogger);
+                //adding usings
+                var newRoot = docBuilder.AddUsings(new CodeChangeOptions());
+                //add code snippets/changes.
+                if (programCsFile.Methods != null && programCsFile.Methods.Any())
+                {
+                    var globalChanges = programCsFile.Methods.Where(x => x.Key.Equals("Global", StringComparison.OrdinalIgnoreCase)).First().Value;
+                    foreach (var change in globalChanges.CodeChanges)
+                    {
+                        change.Block = string.Format(change.Block, mapMethodName);
+                        newRoot = DocumentBuilder.ApplyChangesToMethod(newRoot, new CodeSnippet[] {change}) as CompilationUnitSyntax;
```

Comment:
Would it make sense to move this out of the foreach block, and call something like "ApplyChangesToMethod(newRoot, updatedCodeChanges") ?

Reply:
fixed yeah.